### PR TITLE
[Debugger] Crash test

### LINKED ISSUE #11385: MonoTests.DebuggerTests.Crash test is flaky
## Steps to Reproduce

1. Run Mono.Debugger.Soft tests
2. Observe flaky behavior of MonoTests.DebuggerTests.Crash

### Current Behavior

https://jenkins.mono-project.com/job/test-mono-pull-request-amd64/20338/
https://jenkins.mono-project.com/job/test-mono-pull-request-amd64-cxx/1872/

```
Mono.Debugger.Soft.VMDisconnectedException : Exception of type 'Mono.Debugger.Soft.VMDisconnectedException' was thrown.

  at Mono.Debugger.Soft.Connection.disconnected_check () [0x00018] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs:1323 
  at Mono.Debugger.Soft.Connection.SendReceive (Mono.Debugger.Soft.Connection+CommandSet command_set, System.Int32 command, Mono.Debugger.Soft.Connection+PacketWriter packet) [0x00009] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs:1602 
  at Mono.Debugger.Soft.Connection.SendReceive (Mono.Debugger.Soft.Connection+CommandSet command_set, System.Int32 command) [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs:1644 
  at Mono.Debugger.Soft.Connection.VM_Dispose () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs:1749 
  at Mono.Debugger.Soft.VirtualMachine.Detach () [0x00000] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/VirtualMachine.cs:139 
  at MonoTests.DebuggerTests.Crash () [0x0006c] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/Mono.Debugger.Soft/Test/dtest.cs:2480 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /mnt/jenkins/workspace/test-mono-pull-request-amd64-cxx/mcs/class/corlib/System.Reflection/MonoMethod.cs:324 
```

### Expected Behavior

No crash.

## On which platforms did you notice this

[ ] macOS
[ x ] Linux
[ ] Windows

**Version Used**: master

---

Disabled the test in a509e43cc12f7414c06447c457ac683a6e529ff5.

### PR DESCRIPTION
**Pull request by thaystg**
Changing the test behavior as discussed with Ludovic, because we already have knowledge about the unreliability of the crash reporter and the test is not about it.
So if in 10 times at least one of them works for now it's good.
Fixes #11385

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 1]
```diff
@@ -2457,34 +2457,43 @@ Event step_out_await (string method, Event e)
...
+					vm.Detach ();
+				} catch (VMDisconnectedException vmDisconnect) { //expected behavior because of unreliability of the crash reporter.
+						success = false;
+				} finally {
+					vm = null;
+				}
 			}
+			if (success) 
+				break;
+			System.Console.WriteLine("tentando");
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Extra printf.

Comment (thaystg) [reply mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Removed, thanks!

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
It might be good to track this one down, the crash reporter is unreliable when dealing with random crashes, this one is a controlled one, so it should work.

**Comment by thaystg** [comment]
> e is a controlled one, so it should work.

What do you suggest?
if (summarizer_supervisor_start (&synch)) {
	success = mono_threads_summarize_execute (ctx, out, hashes, silent, mem, provided_size);
	summarizer_supervisor_end (&synch);
} else {
	summarizer_supervisor_wait (&synch);
}

What happens is that after the fork the parent process disappear(between the fork and this call mono_threads_summarize_execute) and only the child process continues.

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Pull request merged**